iron-overlay-behavior.html

```diff
@@ -0,0 +1,470 @@
+positioned via setting `right` and `top`, set dimensions.position to an
+object like this: `{v: 'top', h: 'right'}`.
+
+@event iron-overlay-position
+@param {Object} detail
+@param {Object} detail.target the overlay target
+@param {Object} detail.sizingTarget the overlay sizing target
+@param {Object} detail.opened the opened state
+-->
+
+<!--
+Fired when the user cancels the `iron-overlay`.
+
+@event iron-overlay-cancel
+@param {Object} detail
+@param {Object} detail.target the overlay target
+-->
+
+<script>
+
+  Polymer.IronOverlayBehavior = [Polymer.IronFitBehavior, {
+
+    properties: {
+
+      /**
+       * Set opened to true to show an overlay and to false to hide it.
+       * A `iron-overlay` may be made initially opened by setting its
+       * `opened` attribute.
+       */
+      opened: {
+        observer: '_openedChanged',
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * True if the overlay was cancelled.
+       */
+      canceled: {
+        readOnly: true,
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * If true, the overlay has a backdrop darkening the rest of the screen.
+       * The backdrop element is attached to the document body and may be styled
+       * with the class `iron-overlay-backdrop`. When opened the `iron-opened`
+       * class is applied.
+       */
+      withBackdrop: {
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * Set to true to disable auto-focusing the overlay or child nodes with
+       * the `autofocus` attribute` when the overlay is opened.
+       */
+      noAutoFocus: {
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * Set to true to disable canceling the overlay with the ESC key.
+       */
+      noCancelOnEscKey: {
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * Set to true to disable canceling the overlay by clicking outside it.
+       */
+      noCancelOnOutsideClick: {
+        type: Boolean,
+        value: false
+      },
+
+      _manager: {
+        type: Object,
+        value: Polymer.IronOverlayManager
+      },
+
+      _boundOnCaptureClick: {
+        type: Function,
+        value: function() {
+          return this._onCaptureClick.bind(this);
+        }
+      },
+
+      _boundOnCaptureKeydown: {
+        type: Function,
+        value: function() {
+          return this._onCaptureKeydown.bind(this);
+        }
+      },
+
+      _boundOnResize: {
+        type: Function,
+        value: function() {
+          return this._onResize.bind(this);
+        }
+      }
+
+    },
+
+    listeners: {
+      'click': '_onClick'
+    },
+
+    get backdropElement() {
+      return this._backdrop;
+    },
+
+    get _focusNode() {
+      return Polymer.dom(this).querySelector('[autofocus]') || this;
+    },
+
+    registered: function() {
+      this._backdrop = document.createElement('iron-overlay-backdrop');
```

Comment:
This is interesting, since this is a behavior, but apparently it also wishes it had a template :crossed_flags: